src/main/scala/ru/otus/sc/greet/service/InvocationCounterService.scala

```diff
@@ -0,0 +1,17 @@
+package ru.otus.sc.greet.service
+
+/**
+ * Сервис подсчёта количества вызвов
+ */
+trait InvocationCounterService extends AppService {
+  def incrementAndGet(name: String): Long
+  def get(name: String): Option[Long]
+
+  /**
+   *
+   * @return имя метода вызованного наиболее количество раз
+   */
+  def top(): Option[String]
```

Comment:
```suggestion
  def top: Option[String]
```

Синтаксис `foo()` как правило применяется для функций где мы хотим обозначить какие-то сайд-эффекты, этот метод очень похож на обычный геттер - вряд ли там будут сайд-эффекты.